### ISSUE TITLE
ImportError: cannot import name 'clipboard' from 'nicegui.functions'

### ISSUE THREAD

**Issue by cleesmith** [marcelotduarte/cx_Freeze#2611]
**Describe the bug**
ImportError: cannot import name 'clipboard' from 'nicegui.functions'

**To Reproduce**
The works perfectly before:
```
python setup.py build
```
... with conda, python 3.10.15, and pip ... but not after the build.

**Expected behavior**
To run the same as in dev.

**Desktop:**
 - Platform information: macOS Sequoia 15.0, macbook pro
 - OS architecture: ARM64 (?) M3 Max
 - cx_Freeze version: 7.2.3
 - Python version: 3.10.15

Using this setup.py:
```
import sys
from cx_Freeze import setup, Executable

build_exe_options = {
    "packages": [
        "os", "sys", "nicegui", "uvicorn", "fastapi", "httpx", "tinydb",
        "multiprocessing", "signal", "traceback", "uuid", "re", "time", "datetime",
        "asyncio", "collections", "pyperclip", "starlette", "webview", "openai",
        "google.generativeai", "anthropic", "groq", "mistralai", "ollama"
    ],
    "include_files": ["images/"],
    "includes": ["nicegui.functions.clipboard"], 
}

# Base should be None for non-Windows platforms
base = None

# Mac-specific app bundle options
bdist_mac_options = {
    "iconfile": "Funnel.icns",
    "bundle_name": "AidetourChat",
}

executables = [
    Executable(
        "aidetour_chat.py",
        base=base,
        target_name="AidetourChat",  # The name of your macOS app
    )
]

setup(
    name="AidetourChat",
    version="1.0.0",
    description="Aidetour Chat is a personal, multi-provider AI chat application that allows you to interact with various AI language models from different providers—all in one place.",
    license="MIT",
    keywords="AI, chat, multi-provider, language models, personal AI",
    options={
        "bdist_mac": bdist_mac_options,
    },
    executables=executables,
)

```

... i see this in Terminal:
```
./build/exe.macosx-11.0-arm64-3.10/AidetourChat 

Traceback (most recent call last):
  File "/opt/miniconda3/envs/ac/lib/python3.10/site-packages/cx_Freeze/initscripts/__startup__.py", line 140, in run
    module_init.run(f"__main__{name}")
  File "/opt/miniconda3/envs/ac/lib/python3.10/site-packages/cx_Freeze/initscripts/console.py", line 25, in run
    exec(code, main_globals)
  File "aidetour_chat.py", line 32, in <module>
    from nicegui import app, ui, native, functions
  File "/opt/miniconda3/envs/ac/lib/python3.10/site-packages/nicegui/__init__.py", line 1, in <module>
    from . import elements, run, ui
  File "/opt/miniconda3/envs/ac/lib/python3.10/site-packages/nicegui/ui.py", line 225, in <module>
    from .functions import clipboard
ImportError: cannot import name 'clipboard' from 'nicegui.functions' (/Users/cleesmith/aidetour-chat/build/exe.macosx-11.0-arm64-3.10/lib/nicegui/functions/__init__.pyc)
```
And I've tried several setup.py's and even:
```
cxfreeze build
```
... to use the: pyproject.toml with the same end result.

**Comment by marcelotduarte** [comment]
I'm using linux to test.
I cannot run using python:
```
python aidetour_chat.py 
[pywebview] GTK cannot be loaded
Traceback (most recent call last):
  File "/home/marcelo/.local/venv/aidetour-chat/lib/python3.10/site-packages/webview/guilib.py", line 37, in import_gtk
    import webview.platforms.gtk as guilib
  File "/home/marcelo/.local/venv/aidetour-chat/lib/python3.10/site-packages/webview/platforms/gtk.py", line 21, in <module>
    import gi
ModuleNotFoundError: No module named 'gi'
[pywebview] QT cannot be loaded
Traceback (most recent call last):
  File "/home/marcelo/.local/venv/aidetour-chat/lib/python3.10/site-packages/webview/guilib.py", line 49, in import_qt
    import webview.platforms.qt as guilib
  File "/home/marcelo/.local/venv/aidetour-chat/lib/python3.10/site-packages/webview/platforms/qt.py", line 24, in <module>
    from qtpy import QtCore
ModuleNotFoundError: No module named 'qtpy'
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python3.10/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/marcelo/.local/venv/aidetour-chat/lib/python3.10/site-packages/nicegui/native/native_mode.py", line 50, in _open_window
    webview.start(storage_path=tempfile.mkdtemp(), **core.app.native.start_args)
  File "/home/marcelo/.local/venv/aidetour-chat/lib/python3.10/site-packages/webview/__init__.py", line 166, in start
    guilib = initialize(gui)
  File "/home/marcelo/.local/venv/aidetour-chat/lib/python3.10/site-packages/webview/guilib.py", line 124, in initialize
    raise WebViewException(
webview.errors.WebViewException: You must have either QT or GTK with Python extensions installed in order to use pywebview.
```
Then, I installed pygobject and it runs.
So I can build with:
`cxfreeze --script aidetour_chat.py --includes=nicegui.functions.clipboard --packages=httplib2 --include-files=image`

But it does not run.
I removed the sys.output and put a print() in latest exception and discovered a error in os._exit() function.
But it does not show window yet.
But, by running it, you will be able to debug it. Perhaps using some prints at points where you expect the code to run, etc.

**Comment by cleesmith** [comment]
## TL;DR 
It works after using a new simpler setup.py file and a new build command with bdist_'s at the end of command.

## Mac
My issue seems to be my original setup.py but this simpler one is working now:

```
import sys
from cx_Freeze import setup, Executable

# Dependencies required by your application
build_exe_options = {
    "packages": [
        "os", "sys", "nicegui", "uvicorn", "fastapi", "httpx", "tinydb",
        "multiprocessing", "signal", "traceback", "uuid", "re", "time", "datetime",
        "asyncio", "collections", "pyperclip", "starlette", "webview", "openai",
        "google.generativeai", "anthropic", "groq", "mistralai", "ollama", "socks"
    ],
    "include_files": ["images/"],  # Add any static files here
}

# Setting the base to None for non-Windows platforms.
# For macOS GUI apps, base is set to None.
base = None

setup(
    name="AidetourChat",
    version="1.0",
    description="Aidetour Chat app",
    options={"build_exe": build_exe_options},
    executables=[Executable("aidetour_chat.py", base=base)],
)
```

Using:
```
poetry run python setup.py build bdist_dmg  (or bdist_mac)
```
... double-click on:
```
build/aidetourchat-1.0.app
```
... works as expected.

## Windows
Next up for me is Windows 10/11 and cx_freeze, and the app already works there.

## Linux
And lastly, Linux, like Ubuntu 24.04 ... and I know there's a lot needed to make it run:

- check python version installed, 10, 11, and 12 should work

- install poetry via curl then do:
```
poetry install ... to install lib's via pyproject.toml and poetry.lock

poetry show ... ensure all lib's are installed

sudo apt install libwebkit2gtk-4.1-dev python3-gi python3-gi-cairo gir1.2-webkit2-4.1

sudo apt install build-essential

sudo apt install libcairo2-dev

sudo apt install python3-dev python3-pkgconfig

sudo apt install libgirepository1.0-dev gobject-introspection

sudo apt install cmake

poetry add PyGObject 
```
... not totally sure about all of the above or the proper order of them ... yet ... but linux needs more just to run the app, and of course it needs to be the desktop version (the app is GUI only) added on top of the server version, well, since I am using UTM for a virtual machine on Mac.

**Comment by cleesmith** [comment]
Thanks for cx_Freeze.

**Comment by marcelotduarte** [comment]
You're welcome.
